agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/Samplers.java

```diff
@@ -21,12 +21,15 @@ public static Sampler getSampler(
       sampler = new AiSampler(requestSamplingPercentage, parentlessDependencySamplingPercentage);
     } else if (sampling.percentage != null) {
       SamplingPercentage samplingPercentage;
-      if (sampling.percentage == 100) {
+      if (sampling.percentage == 100 && !Boolean.FALSE.equals(sampling.ingestion)) {
```

Comment:
At line 173, `ingestion` may be `FALSE` (`Boolean ingestion = Boolean.FALSE)`? In this way, line 24 could become 
```suggestion
      if (sampling.percentage == 100 && Boolean.TRUE.equals(sampling.ingestion)) {
```
Which could be easier to read

Reply:
this code is gone now after rewrite